contracts/mocks/ERC4626ReentrantAsset.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../token/ERC20/ERC20.sol";
+import "../token/ERC20/extensions/ERC4626.sol";
+
+contract ERC4626ReentrantAsset is ERC20("TEST", "TST") {
```

Comment:
Can be find a better name? This is not ERC4626.

Sure it is used in the context of ERC4626, but its more of a `ERC20ArbitraryCallMock` or something like that.